### DISCUSSION TITLE
[DNS Client] doge (DNS client like dig in rust)

### DISCUSSION THREAD

**Discussion by cocoon** [mokeyish/smartdns-rs#430] [General]
I have just seen the commit for "Add support for resolve subcommand (similar to dig)" and thought I just mention that there is a rust based dns client available here:

https://github.com/Dj-Codeman/doge

doge is a command-line DNS client, like dig forked from the amazing work done [here](https://github.com/ogham/dog). It has colourful output, understands normal command-line argument syntax, supports the DNS-over-TLS and DNS-over-HTTPS protocols, and can emit JSON. I believe this is an amazing project and should be improved on.

**Comment by mokeyish** [comment]
Thanks, I used to use dog https://github.com/ogham/dog , but it seems it hasn't been updated for a long time.

So, I think it is not a big deal to add a subcommand, so I submit a simple implementation version first.

I will consider adding more display formats like dog.

**Reply by mokeyish** [reply]
https://github.com/mokeyish/smartdns-rs/pull/423

This PR supports adding soft links. There is no dig on Windows. Creating a dig soft link will give you a similar experience to dig on Linux.

**Reply by cocoon** [reply]
Yes dogE is the imporved maintained version of dog.
That's why I thought I would mention it here. 👍
